File: shoring/desk/src/main/java/com/panopset/marin/game/card/images/BlackjackImages.java
package com.panopset.marin.game.card.images;

import java.net.URL;
import javafx.scene.image.Image;

public enum BlackjackImages {

  INSTANCE;

  public static Image getFxArrowImg() {
    return INSTANCE.getArrowImg();
  }

  private Image arrowImg;

  private Image getArrowImg() {
    if (arrowImg == null) {
      arrowImg = new Image(getClass().getResource(ARROW_IMAGE).toExternalForm());
    }
    return arrowImg;
  }

  private Image cardMap;

  public static Image getCardMap() {
    return INSTANCE.getFxCardMap();
  }

  private Image getFxCardMap() {
    if (cardMap == null) {
      cardMap = new Image(getClass().getResource(DFT_CARD_IMAGE).toExternalForm());
    }
    return cardMap;
  }

  private Image chipMap;

  public static Image getChipMap() {
    return INSTANCE.getFxChipMap();
  }

  private Image getFxChipMap() {
    if (chipMap == null) {
      URL url = getClass().getResource(DFT_CHIP_IMAGE);
      String imgextf = url.toExternalForm();
      chipMap = new Image(imgextf);
    }
    return chipMap;
  }

  /**
   * Default card image is the one from jfitz.com.
   */
  private final String DFT_CARD_IMAGE = "/com/panopset/marin/game/card/images/jfitzcards.png";


  /**
   * Panopset owned image, created by Karl Dinwiddie 2012-05-19.
   */
  private final String DFT_CHIP_IMAGE = "/com/panopset/marin/game/card/images/chips.png";

  /**
   * Panopset owned image.
   */
  private final String ARROW_IMAGE = "/com/panopset/marin/game/card/images/arrow.png";

}

File: code/shoring/desk/src/main/kotlin/com/panopset/marin/game/card/images/BlackjackImages.kt
package com.panopset.marin.game.card.images

import javafx.scene.image.Image

object BlackjackImages {
    /**
     * Default card image is the one from jfitz.com.
     */
    private const val DFT_CARD_IMAGE = "/com/panopset/marin/game/card/images/jfitzcards.png"


    /**
     * Panopset owned image, (which means it is open source) created by Karl Dinwiddie 2012-05-19.
     */
    private const val DFT_CHIP_IMAGE = "/com/panopset/marin/game/card/images/chips.png"

    /**
     * Panopset owned image.
     */
    private const val ARROW_IMAGE = "/com/panopset/marin/game/card/images/arrow.png"

    val arrowImg = Image(javaClass.getResource(ARROW_IMAGE)?.toExternalForm())
    val cardMap = Image(javaClass.getResource(DFT_CARD_IMAGE)?.toExternalForm())
    val chipMap = Image(javaClass.getResource(DFT_CHIP_IMAGE)?.toExternalForm())
}
